src/Build/BackEnd/Components/BuildRequestEngine/FullyQualifiedBuildRequest.cs

```diff
@@ -25,68 +23,42 @@ namespace Microsoft.Build.BackEnd
     internal class FullyQualifiedBuildRequest
     {
         /// <summary>
-        /// The request's configuration.
-        /// </summary>
-        private BuildRequestConfiguration _requestConfiguration;
-
-        /// <summary>
-        /// The set of targets to build.
-        /// </summary>
-        private string[] _targets;
-
-        /// <summary>
-        /// Whether or not we need to wait for results before completing this request.
-        /// </summary>
-        private bool _resultsNeeded;
-
-        /// <summary>
         /// Initializes a build request.
         /// </summary>
         /// <param name="config">The configuration to use for the request.</param>
         /// <param name="targets">The set of targets to build.</param>
         /// <param name="resultsNeeded">Whether or not to wait for the results of this request.</param>
-        public FullyQualifiedBuildRequest(BuildRequestConfiguration config, string[] targets, bool resultsNeeded)
+        /// <param name="flags">Flags specified for the build request.</param>
+        public FullyQualifiedBuildRequest(BuildRequestConfiguration config, string[] targets, bool resultsNeeded, BuildRequestDataFlags flags = BuildRequestDataFlags.None)
```

Comment:
Maybe just send the bool to the FullyQualifiedBuildRequest? I don't think it needs the entire enum right?